test/integration/component/test_secsr_mount.py

```diff
@@ -0,0 +1,170 @@
+
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+        return
+
+
+    def isOnlyLocalStorageAvailable(self):
+        if not(self.zone.localstorageenabled):
+            return False
+        
+        storage_pools = StoragePool.list(
+                   self.apiclient,
+                   zoneid=self.zone.id,
+                   listall=True
+                    )
+        self.assertEqual(
+                           isinstance(storage_pools, list),
+                           True,
+                           "Check if listStoragePools returns a valid response"
+                           )
+        for storage_pool in storage_pools:
+            if storage_pool.type == u'NetworkFilesystem':
+                return False
+            
+        return True
+
+    def areDownloaded(self, id):
+        # Fetch details of templates copied from table in database
+        qresultset = self.dbclient.execute(
+            "SELECT template_id, download_state FROM cloud.template_spool_ref, cloud.vm_template where \
+            cloud.template_spool_ref.template_id=cloud.vm_template.id and cloud.vm_template.uuid='%s';" %
+            id)
+        self.logger.debug("Qresult %s" % format(qresultset))
+        if not isinstance(qresultset, list) or len(qresultset) <= 1:
+            return False, 1
+        
+        for result in qresultset:
+            self.logger.debug('Template %s download status %s' % (result[0], result[1])) 
+            if result[1] != 'DOWNLOADED':
+                return False, 1
+            
+        return True, 1
+
+    @attr(
+        tags=[
+            "advanced",
+            "xenserver"],
+        required_hardware="true")
+    def test_01_prepare_template_local_storage(self):
+    
+        if not(self.isOnlyLocalStorageAvailable()):
+            raise unittest.SkipTest("Skipping this test as this is for Local storage on only.");
+            return
+        
+        listHost = Host.list(
+            self.apiclient,
+            type='Routing',
+            zoneid=self.zone.id,
+            podid=self.pod.id,
+        )
+        for host in listHost:
+            self.logger.debug('Host id %s, hypervisor %s, localstorage %s' % (host.id, host.hypervisor, host.islocalstorageactive))
+                  
+        if len(listHost) < 2:
+            self.logger.debug("Prepare secondary storage race condition can be tested with two or more host only %s, found" % len(listHost));
+            raise unittest.SkipTest("Prepare secondary storage can be tested with two host only %s, found" % len(listHost));
+            return
+        
+        list_template_response = Template.list(
+                                            self.apiclient,
+                                            templatefilter='all',
+                                            zoneid=self.zone.id)
+        
+        template_response = list_template_response[0]
```

Comment:
To avoid errors when nothing is returned, this may call on a None[0] causing NoneType errors. Check that this is a valid list first, using something like instance of list?

Reply:
fixed